Lib/test/test_tempfile.py

```diff
@@ -957,6 +957,12 @@ def test_bad_mode(self):
             tempfile.NamedTemporaryFile(mode=2, dir=dir)
         self.assertEqual(os.listdir(dir), [])
 
+    def test_if_temp_file_already_removed(self):
+        tmpdir = tempfile.mkdtemp()
```

Comment:
Should add a self.addCleanup on that tmpdir—see the test above (`test_bad_mode`) for an example, just to make sure that if this test ever does fail that we don't have a temp dir that doesn't get cleaned up by us.